roles/core_configure/tasks/removenode.yml

```diff
@@ -29,7 +29,7 @@
 
     - name: delete | Set delete node
       set_fact:
-        delete_nodes: "{{ groups['scale_remove_nodes'] | list }}"
+        delete_nodes: "{{ groups['scale_remove_nodes'] | map('extract', hostvars, 'scale_daemon_nodename') | list }}"
```

Comment:
@acch I posted above mentioned comment earlier . Thank you!

Reply:
Sorry Rajan, not sure why I missed your comment before.

`scale_daemon_nodename` should always be defined for every inventory host, as it is defined in `core_common/defaults/main.yml`. This role is a dependency to all other roles, so the variable should always be defined... I can't think of a scenario where it's not defined for an inventory host.

What can happen, though, is that if one adds hosts to groups using a "wrong" name - i.e. a name that's different from the inventory hostname. In that case Ansible will not reference the (existing) inventory host, which has all default variables initialized. Instead, it will create a new host - one that has no corresponding entry in the inventory. Hence, this host will not have any variables defined. This may lead to all sorts of strange errors, such as the one originally reported here:

```shell
...'ansible.vars.hostvars.HostVarsVars object' has no attribute 'scale_cluster_gui'...
```

I could imagine that what you're referring to (`scale_install_updated` not defined) has a similar root cause.

Does this make sense?

Reply:
@acch Thank you! Yes correct